hnn_core/network_builder.py

```diff
@@ -366,11 +366,13 @@ def _gid_assign(self):
             # get list of all NetworkDrives that contact this cell, and
             # make sure the corresponding _ArtificialCell gids are associated
             # with the current node/rank
-            for key in self.net._p_unique.keys():
-                gid_input = gid + self.net.gid_ranges[key][0]
-                _PC.set_gid2node(gid_input, rank)
-                self._gid_list.append(gid_input)
+            for key in self.net.external_drives.keys():
+                if key != 'common':
+                    gid_input = gid + self.net.gid_ranges[key][0]
+                    _PC.set_gid2node(gid_input, rank)
+                    self._gid_list.append(gid_input)
 
+        # XXX must redo for new drives API
```

Comment:
Still thinking about how to avoid all that `!=`-mess... Thinking it might be good to have a reverse-lookup method for

```python
drives_gid_list = net.this_cell_drives(gid)
for gid_input in drives_gid_list:
    _PC.set_gid2node(gid_input, rank)
...
```